queries.sql

```diff
@@ -79,3 +79,69 @@ SELECT COUNT(*), owners.full_name
    ON animals.owner_id = owners.id
    GROUP BY owners.full_name
    ORDER BY COUNT DESC LIMIT 1;
+
+SELECT vets.name, animals.name , date_of_visit
+   FROM animals JOIN visits
+   ON animals.id = visits.animals_id
+   JOIN vets
+   ON vets.id = visits.vets_id 
+   WHERE vets.name = 'William Tatcher'
+   ORDER BY date_of_visit DESC LIMIT 1;
+
+SELECT vets.name, COUNT(visits.date_of_visit) 
+   FROM visits JOIN vets
+   ON vets.id = visits.vets_id
+   WHERE vets.name = 'Stephanie Mendez';
+   GROUP BY vets.name;
+
+SELECT vets.name, specializations.species_id, specializations.vets_id, species.name
+   FROM specializations FULL OUTER JOIN species
+   ON specializations.id = species.id
+   FULL OUTER JOIN vets
+   ON specializations.vets_id = vets.id;
```

Comment:
- [ Required ] I think this query does not return a correct result. In the screenshot, it lists _Stephanie Mendez_ and _Jack Harkness_ as vets with no specialty which is wrong since they specialize in `Pokemon` & `Digimon` respectively. I would like to suggest that you refactor this query as shown below so that a correct result is returned.

## Example code
```SQL
  SELECT vets.*, species.name
  FROM vets
  LEFT JOIN specializations ON vets.id = specializations.vet_id
  LEFT JOIN species ON specializations.species_id = species.id;

```

## Current result set
![](https://user-images.githubusercontent.com/67557999/253727951-86e904c6-4358-4052-ab3a-fdb991e52c0c.png)